File: chrome/test/data/webui/cr_components/cr_components_mojo_browsertest.js
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/** @fileoverview Tests for shared Polymer 3 components using Mojo. */

// Polymer BrowserTest fixture.
GEN_INCLUDE(['//chrome/test/data/webui/polymer_browser_test_base.js']);

GEN('#include "build/build_config.h"');
GEN('#include "components/history_clusters/core/features.h"');
GEN('#include "content/public/test/browser_test.h"');

/** Test fixture for shared Polymer 3 components using Mojo. */
var CrComponentsMojoBrowserTest = class extends PolymerTest {
  /** @override */
  get browsePreload() {
    throw new Error('this is abstract and should be overriden by subclasses');
  }
};

var CrComponentsCustomizeThemesTest =
    class extends CrComponentsMojoBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=cr_components/customize_themes_test.js';
  }
};

TEST_F('CrComponentsCustomizeThemesTest', 'All', function() {
  mocha.run();
});

var CrComponentsHelpBubbleMixinTest =
    class extends CrComponentsMojoBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=cr_components/help_bubble_mixin_test.js';
  }
};

TEST_F('CrComponentsHelpBubbleMixinTest', 'All', function() {
  mocha.run();
});

var CrComponentsHelpBubbleTest = class extends CrComponentsMojoBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=cr_components/help_bubble_test.js';
  }
};

TEST_F('CrComponentsHelpBubbleTest', 'All', function() {
  mocha.run();
});

var CrComponentsHistoryClustersTest =
    class extends CrComponentsMojoBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://history/test_loader.html?module=cr_components/history_clusters_test.js';
  }

  /** @override */
  get featureList() {
    return {
      enabled: [
        'history_clusters::internal::kJourneysImages',
      ],
    };
  }
};

TEST_F('CrComponentsHistoryClustersTest', 'All', function() {
  mocha.run();
});

var CrComponentsMostVisitedTest = class extends CrComponentsMojoBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=cr_components/most_visited_test.js';
  }
};

TEST_F('CrComponentsMostVisitedTest', 'General', function() {
  runMochaSuite('General');
});

TEST_F('CrComponentsMostVisitedTest', 'Layouts', function() {
  runMochaSuite('Layouts');
});

TEST_F('CrComponentsMostVisitedTest', 'LoggingAndUpdates', function() {
  runMochaSuite('LoggingAndUpdates');
});

// crbug.com/1226996
GEN('#if BUILDFLAG(IS_LINUX) && !defined(NDEBUG)');
GEN('#define MAYBE_Modification DISABLED_Modification');
GEN('#else');
GEN('#define MAYBE_Modification Modification');
GEN('#endif');
TEST_F('CrComponentsMostVisitedTest', 'MAYBE_Modification', function() {
  runMochaSuite('Modification');
});

TEST_F('CrComponentsMostVisitedTest', 'DragAndDrop', function() {
  runMochaSuite('DragAndDrop');
});

TEST_F('CrComponentsMostVisitedTest', 'Theming', function() {
  runMochaSuite('Theming');
});

File: chrome/test/data/webui/cr_components/cr_components_mojo_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/test/scoped_feature_list.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "components/history_clusters/core/features.h"
#include "content/public/test/browser_test.h"

typedef WebUIMochaBrowserTest CrComponentsCustomizeColorSchemeModeTest;
IN_PROC_BROWSER_TEST_F(CrComponentsCustomizeColorSchemeModeTest, All) {
  set_test_loader_host(chrome::kChromeUICustomizeChromeSidePanelHost);
  RunTest("cr_components/customize_color_scheme_mode_test.js", "mocha.run()");
}

typedef WebUIMochaBrowserTest CrComponentsCustomizeThemesTest;
IN_PROC_BROWSER_TEST_F(CrComponentsCustomizeThemesTest, All) {
  set_test_loader_host(chrome::kChromeUINewTabPageHost);
  RunTest("cr_components/customize_themes_test.js", "mocha.run()");
}

typedef WebUIMochaBrowserTest CrComponentsHelpBubbleMixinTest;
IN_PROC_BROWSER_TEST_F(CrComponentsHelpBubbleMixinTest, All) {
  set_test_loader_host(chrome::kChromeUINewTabPageHost);
  RunTest("cr_components/help_bubble_mixin_test.js", "mocha.run()");
}

typedef WebUIMochaBrowserTest CrComponentsHelpBubbleTest;
IN_PROC_BROWSER_TEST_F(CrComponentsHelpBubbleTest, All) {
  set_test_loader_host(chrome::kChromeUINewTabPageHost);
  RunTest("cr_components/help_bubble_test.js", "mocha.run()");
}

class CrComponentsHistoryClustersTest : public WebUIMochaBrowserTest {
 protected:
  CrComponentsHistoryClustersTest() {
    scoped_feature_list_.InitAndEnableFeature(
        history_clusters::internal::kJourneysImages);
    set_test_loader_host(chrome::kChromeUIHistoryHost);
  }

 private:
  base::test::ScopedFeatureList scoped_feature_list_;
};

IN_PROC_BROWSER_TEST_F(CrComponentsHistoryClustersTest, All) {
  RunTest("cr_components/history_clusters_test.js", "mocha.run()");
}

class CrComponentsMostVisitedTest : public WebUIMochaBrowserTest {
 protected:
  CrComponentsMostVisitedTest() {
    set_test_loader_host(chrome::kChromeUINewTabPageHost);
  }
};

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, General) {
  RunTest("cr_components/most_visited_test.js", "runMochaSuite('General');");
}

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, Layouts) {
  RunTest("cr_components/most_visited_test.js", "runMochaSuite('Layouts');");
}

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, ReflowLayouts) {
  RunTest("cr_components/most_visited_test.js",
          "runMochaSuite('Reflow Layouts');");
}

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, LoggingAndUpdates) {
  RunTest("cr_components/most_visited_test.js",
          "runMochaSuite('LoggingAndUpdates');");
}

// crbug.com/1226996
#if BUILDFLAG(IS_LINUX) && !defined(NDEBUG)
#define MAYBE_Modification DISABLED_Modification
#else
#define MAYBE_Modification Modification
#endif
IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, MAYBE_Modification) {
  RunTest("cr_components/most_visited_test.js",
          "runMochaSuite('Modification');");
}

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, DragAndDrop) {
  RunTest("cr_components/most_visited_test.js",
          "runMochaSuite('DragAndDrop');");
}

IN_PROC_BROWSER_TEST_F(CrComponentsMostVisitedTest, Theming) {
  RunTest("cr_components/most_visited_test.js", "runMochaSuite('Theming');");
}
